### PR TITLE
chore(vamm): Move code to new repository

### PR DESCRIPTION
**Pull request by Cardosaum**
# Overview

Import the changes made in the branches [`escher-feature/twap-module`](https://github.com/ComposableFi/composable/pull/1597) and [`escher-feature/code-review-swap-simulation`](https://github.com/ComposableFi/composable/pull/1401).

Both branches are being imported at the same time because in order to address this [`comment`](https://github.com/ComposableFi/composable/pull/1401#discussion_r941092581) made by @0xangelo, the Twap module should have been implemented.

Now the two remaining comments in the old PR should be solved ([`#1`](https://github.com/ComposableFi/composable/pull/1401#discussion_r941092581) and [`#2`](https://github.com/ComposableFi/composable/pull/1401#discussion_r941118011)).

I reproduce below the description of both PRs:

---

## Original PR: Implement code review suggestions

This PR implement changes proposed in [`#1372`](https://github.com/ComposableFi/composable/pull/1372). The list of major changes can be listed as:
- Assert runtime storage doesn't change in a `swap_simulation` test.
- Refactor tests for `swap` and `swap_simulation` functionality, using pre-defined constants over hard-coded values directly used in tests.
- Use an already existing method in `FixedPointNumber` over a custom function.
- Remove unneeded error variant.
- Implement `try_update_twap` function, fixing a hack in `swap` function.
- Use `with_context` blocks in tests, removing duplicated code and improving readability.
- Rename fallible traits in `composable-maths` to `Try`-variants, following rust code convention.
- Implement a new trait in `composable-maths` meant to convert `U256` values.
- Remove custom conversion functions in Vamm Pallet, using `composable-maths` equivalent methods.
- Improve `swap_simulation` documentation.
- Fix several broken links in modules `pallet-vamm`, `composable-traits`, `composable-maths` and `composable-support`
- Fix a bug in `compute_swap_base_asset` where removing an asset was not considered `negative`.

---

## Original PR: Twap Module

This PR creates a separate module for the Twap computation.

### CODE REVIEW

File: helpers/src/twap/tests/unit.rs [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+#[case(1 * DAY, 12)]
+#[case(2 * DAY, 12)]
+#[case(3 * DAY, 12)]
+#[case(4 * DAY, 12)]
+#[case(5 * DAY, 12)]
+#[case(6 * DAY, 12)]
+#[case(7 * DAY, 12)]
+#[ignore = "TODO: Still need to configure flag to run this test as this \
+            one takes quite a while and is not meant to be run every time"]
+fn test_polars(#[case] period: u64, #[case] dataset: i32) {
```
Comment (0xangelo) [inline helpers/src/twap/tests/unit.rs:335]: Can't give too much feedback on this, but what are you testing for in general terms?

**Review by 0xangelo — APPROVED** [review]
LGTM, congrats!

**Review by Matteocolella1995 — APPROVED** [review]

File: helpers/src/numbers/tests/signed_math.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        } {
+            (Ok(a), Some(b)) => {
+                assert_eq!(a, b);
+            },
+            (Err(_), None) => (),
+            _ => {
+                panic!("This combination should never happen.");
+            },
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Cardosaum) [reply helpers/src/numbers/tests/signed_math.rs]: Oh, okay.
Fixed that in a later commit :+1:

File: helpers/src/twap/mod.rs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+        rustdoc::missing_crate_level_docs,
+        rustdoc::missing_doc_code_examples,
+        warnings,
+    )
+)]
+#![allow(clippy::wildcard_imports)]
+
+#[cfg(test)]
+mod tests;
+
```
[the remark this answers is not in the document]
Comment (Cardosaum) [reply helpers/src/twap/mod.rs:65]: Not that I'm aware of.
But for consistency's sake, I'll move the declaration :+1:

File: helpers/src/twap/mod.rs [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+};
+use sp_std::cmp::Ord;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// The [`Twap`] value itself, storing both the underlying time weighted average
+/// price and its most recent timestamp.
+///
+/// The [`Twap`] type implement some convinience methods to facilitate when
```
[the remark this answers is not in the document]
Comment (Cardosaum) [reply helpers/src/twap/mod.rs:82]: Thanks!

File: vamm/src/lib.rs [hunk 5]
```diff
@@ -540,9 +546,18 @@ pub mod pallet {
         /// * [`Pallet::do_update_twap`]
         /// * [`Pallet::sanity_check_before_update_twap`]
         NewTwapValueIsZero,
+        /// Tried to update twap value, but a function call responsible to
+        /// return a new twap value didn't do so. As the called function should
+        /// return a value always, not doing so must be an error.
+        ///
```
[the remark this answers is not in the document]
Comment (Cardosaum) [reply vamm/src/lib.rs:552]: Thanks!

File: helpers/src/twap/tests/unit.rs (hunk 1 above)
Comment (0xangelo) [reply helpers/src/twap/tests/unit.rs:335]: Yeah, it makes sense having a separate repo in the organization for data analysis, I agree

**Pull request merged**